code/game/area/area_power.dm

```diff
@@ -23,7 +26,7 @@
 
 // called when power status changes
 /area/proc/power_change()
-	for(var/obj/machinery/M in src)	// for each machine in the area
+	for(var/obj/machinery/M in machinery_list)	// for each machine in the area
```

Comment:
If you use the weakref suggestion below, this will need to be changed to var/weakref. Either way should add `AS_ANYTHING` before `in` as the list will only contain strictly typed objects.